### DISCUSSION TITLE
atal: unable to access 'https://github.com/ramakrisha/ecommerce.git/': Failed to connect to 195.35.15.217 port 1080 after 21077 ms: Couldn't connect to server

### DISCUSSION THREAD

**Discussion by kalidasupiriya** [community/community#122951] [Programming Help] [answered]
### Select Topic Area

Question

### Body

atal: unable to access 'https://github.com/ramakrisha/ecommerce.git/': Failed to connect to 195.35.15.217 port 1080 after 21077 ms: Couldn't connect to server

**Comment by CopperEagle** [comment]
Interesting @kalidasupiriya 

First things first, checking the network side. [This](https://raw.githubusercontent.com/hostinger/geofeed/main/geofeed.csv) and [this](https://www.whois.com/whois/195.35.15.217) confirm `195.35.15.217` is indeed a Github IP, so there is no issue with your DNS lookup. Also, according to [the Github status page](https://www.githubstatus.com/), the Github systems are all operational. Pinging the server, it was up.
There could be a problem with your network setup. Are you using a VPN? Also, try to clone any other public repository using `git clone <repo>`. Does it work?

The other thing is, you need to use tokens to authenticate. Can't use the password. Do you have a token? Check [this](https://unix.stackexchange.com/questions/519111/unable-to-access-github-couldnt-connect-to-server) for a similar situation and how to setup a token (second answer).

**Reply by kalidasupiriya** [reply]
this is not this server ip address it's another site ip this time i'm not able to clone also   (this is my repo https://github.com/ramakrisha/ecommerce.git/')

**Reply by CopperEagle** [reply]
@kalidasupiriya The repository you mention is private. So when I try to clone it, I am asked to authenticate (you should be, too).
 Could you try the fallowing line? `git clone https://github.com/ramakrisha/yiitest.git` This is one of your public repos. If it works, then the problem is with authentication. If not, it could be with the configuration. Again, you may be interested in [this](https://unix.stackexchange.com/questions/519111/unable-to-access-github-couldnt-connect-to-server).

**Reply by kalidasupiriya** [reply]
git clone https://github.com/ramakrisha/yiitest.git this one also not connected

Fatal: unable to access 'https://github.com/ramakrisha/ecommerce.git/': Failed to connect to 195.35.15.217 port 1080 after 21077 ms: Couldn't connect to server
when comes to this error  https://github.com/ramakrisha/ecommerce.git/ this is my git repository last two moths also working on this repo
and recently im connected   195.35.15.217 is this is another site ip (because i want to connect that ip to my git  but that was not connected properly) at this time my old repo is showing this Fatal  error  can u tell me what the issue is here, how to fix it.

**Reply by CopperEagle** [reply]
Must be network.

A good first try is to check the firewall. Note that the website Gthub is a different IP than the content servers. So you being able to see this site but cannot connect to the content server is possible. One fast jab to solve this would be to use a VPN and see if it works.

A sidenote, though I do not think it applies to you. People in some countries may likely have trouble accessing these sites, due to externally operated firewalls operated by govs.

Edit: Moved comment over to be a reply.

**Comment by kalidasupiriya** [comment, marked as answer]
The error message indicates that Git is trying to connect through a proxy server at 195.35.15.217 on port 1080, which is not your intended destination. It appears that your Git configuration is set to use this proxy, which is either incorrect or misconfigured. To resolve this issue, you need to update or remove the proxy settings in your Git configuration.

Here are the steps to fix this:
Step 1:
Check Your Git Proxy Settings
Open a terminal or command prompt.
Check your global Git proxy settings by running the following commands:

git config --global --get http.proxy
git config --global --get https.proxy

If a proxy is configured, these commands will return the proxy settings.

Step 2: Remove or Update Proxy Settings
Remove Proxy Settings
If you do not need a proxy to connect to GitHub, remove the proxy settings
1:Run the following commands to unset the proxy:

git config --global --unset http.proxy
git config --global --unset https.proxy

when the the proxy is unset my git connection is working properly
